backend/src/apis/google-natural-lang-api.ts

```diff
@@ -30,30 +30,34 @@ const callGoogleSenti = function(inputText: string) {
     };
   }
 
-  const calculateSentimentAverage = function(entities: IEntity[]): number | null {
-    if (entities.length === 0) {
-      return null;
-    }
+  const calculateSentimentAverage = function (entities: IEntity[]): Promise<number> {
 
-    let totalScore = 0;
+    return new Promise((resolve, _reject) => {
+      if (entities.length === 0) {
+        return null;
+      }
 
-    for (const entity of entities) {
-      totalScore += entity.sentiment.score;
-    }
+      let totalScore = 0;
 
-    const averageScore = totalScore / entities.length;
+      for (const entity of entities) {
+        totalScore += entity.sentiment.score;
+      }
+
+      const averageScore = totalScore / entities.length;
+      resolve(averageScore);
+    });
```

Comment:
I dont think this needs to be a promise because `languageClient.analyzeEntitySentiment` will return a promise already of the value inside its `.then`

Reply:
I keep forgetting that if you return something inside a then block it wraps it in a promise as well.